frontend/src/common/MultiFileInput.tsx

```diff
@@ -0,0 +1,89 @@
+import { Box, Chip } from '@mui/material'
+import Button from '@mui/material/Button'
+import { styled } from '@mui/system'
+import { ChangeEvent, Dispatch, SetStateAction } from 'react'
+
+const Input = styled('input')({
+  display: 'none',
+})
+
+const displayFilename = (filename: string) => {
+  const parts = filename.split('.')
+  const ext = parts.pop()
+  const base = parts.join('.')
+  return base.length > 12 ? `${base}...${ext}` : filename
+}
+
+export default function MultiFileInput({
+  label,
+  setFiles,
+  files,
+  accepts,
+  disabled,
+  fullWidth = false,
+}: {
+  label: string
+  setFiles: Dispatch<SetStateAction<File[] | undefined>>
+  files?: File[]
+  accepts?: string
+  disabled?: boolean
+  fullWidth?: boolean
+}) {
+  const id = `${label.replace(/ /g, '-').toLowerCase()}-file`
+
+  function handleDelete(fileToDelete: File) {
+    if (files) {
+      const updatedFileList = files.filter((file) => file.name !== fileToDelete.name)
+      setFiles(updatedFileList)
+    }
+  }
+
+  function handleFileChange(event: ChangeEvent<HTMLInputElement>) {
+    const newFiles = event.target.files ? Array.from(event.target.files) : []
+    if (newFiles) {
+      if (files) {
+        const updatedFiles = newFiles.concat(
+          files.filter((file) => newFiles.every((newFile) => newFile.name != file.name))
```

Comment:
Slight performance improvement could be made here. Instead of always checking `every` new file we can exit early using `some`:
```
newFiles.concat(
          files.filter((file) => !newFiles.some((newFile) => newFile.name === file.name))
)
```